src/reducers/index.ts

```diff
@@ -437,6 +435,13 @@ const appStateReducer = (state: AppState = defaultState, action: Actions): AppSt
             Debug.log('App state set');
             return action.payload.appState;
         }
+        case 'APP-STATE-UPDATE-FUNCTION': {
+            const stateCopy = {...state};
+            const partialAppState = action.payload.appStateUpdater(stateCopy);
+            Object.assign(stateCopy, partialAppState);
```

Comment:
this could be {
   ...stateCopy,
   ...partialAppState,
}

Reply:
Or even 
```
return {
   ..state,
   ...partialAppState,
}
```